### PR TITLE
[cc] Encrypt usernames

### PR DESCRIPTION
**Pull request by silversupreme**

### CODE REVIEW

File: callisto_core/accounts/auth.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+from django.conf import settings
+from django.contrib.auth.hashers import check_password
+from django.contrib.auth.models import User
+
+from callisto_core.accounts.models import Account
+
+
+def index(src, length=8):
+    """Computes an index suitable for fast lookups on a user record."""
+    base = f"{src}//{settings.INDEXING_KEY}"
```
Comment (e-lisa) [inline callisto_core/accounts/auth.py]: This should also be configured in `callisto_core/utils/settings.py` with `os.getenv()` (examples of such are in the file)

File: callisto_core/accounts/migrations/0004_encrypt_user_data.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        account.save()
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [("accounts", "0003_auto_20190607_1540")]
+
+    operations = [
+        migrations.RunPython(
+            encrypt_user_data, reverse_code=migrations.RunPython.noop
```
[the remark this answers is not in the document]
Comment (silversupreme) [reply callisto_core/accounts/migrations/0004_encrypt_user_data.py]: re-applying will re-compute anyways (which bit me a time or two during testing)

[the remark this answers is not in the document]
Comment (e-lisa) [reply callisto_core/accounts/migrations/0004_encrypt_user_data.py]: fair enough. i will resolve this one

File: callisto_core/accounts/auth.py (hunk 1 above)
Comment (e-lisa) [reply callisto_core/accounts/auth.py]: To clarify, by "this" i mean "settings.INDEXING_KEY"

**Pull request merged**